mlrun/api/db/sqldb/db.py

```diff
@@ -1181,6 +1181,36 @@ def list_entities(
                 )
         return schemas.EntitiesOutput(entities=entities_results)
 
+    @staticmethod
+    def _assert_group_by_parameters(group_by, sort):
+        if sort is None:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "sort parameter must be provided when group_by is used."
+            )
+        # For now, name is the only supported value. Remove once more fields are added.
+        if group_by != schemas.GroupByField.name:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"group_by for feature-store objects must be 'name'. Value given: '{group_by.value}'"
+            )
+
+    @staticmethod
+    def _create_partitioned_query(session, query, cls, group_by, order, rows_per_group):
+        row_number_column = (
+            func.row_number()
+            .over(
+                partition_by=group_by.to_group_by_db_field(cls),
+                order_by=order.to_order_by_predicate(cls.updated),
+            )
+            .label("row_number")
+        )
+
+        # Need to generate a subquery so we can filter based on the row_number, since it
+        # is a window function using over().
+        subquery = query.add_column(row_number_column).subquery()
+        return session.query(aliased(cls, subquery)).filter(
+            subquery.c.row_number <= rows_per_group
+        )
```

Comment:
I hope you tested it well, cause that is Chinese to me 😅

Reply:
It's not THAT complicated, will be happy to walk you through the details. And, yes.